internal/wasi/fs_test.go

```diff
@@ -12,10 +17,242 @@ import (
 func TestMemFile_Read_Seek(t *testing.T) {
 	expectedFileContent := []byte("wazero") // arbitrary contents
 	memFile := &memFile{
-		buf: expectedFileContent,
+		fsEntry: &memFSEntry{Contents: expectedFileContent},
 	}
 	// TestReader tests that io.Reader correctly reads the expected contents.
 	// It also tests io.Seeker when it's implemented, which memFile does.
 	err := iotest.TestReader(memFile, expectedFileContent)
 	require.NoError(t, err)
 }
+
+// TestMemFS_FSInterface tests the behavior of fs.FS interface of MemFS by fstest.TestFS.
+// It includes testing the behavior of fs.FS.Open, and fs.File.(Read, Seek, Stat, ReadDir)
+// See io/fs.FS
+// See io/fs.File
+// See fstest.TestFS
+func TestMemFS_FSInterface(t *testing.T) {
+	memFS := &MemFS{
+		"simple file": {Contents: []byte("simple cont")},
+		"directory": {
+			Entries: map[string]*memFSEntry{
+				"file1 inside directory": {Contents: []byte("simple cont file1 inside directory")},
```

Comment:
maybe use more realistic paths here? it would make it easier to read and know the intent.

ex would it be "directory/file1" because this includes the name of the key?

also the contents if dummy can be shorter like single character or single word

Reply:
I've changed the names to more realistic paths, but let me know if those are not what you intended. THanks!